src/vcpkg/binarycaching.cpp

```diff
@@ -2310,137 +1856,393 @@ namespace
             };
 
             MetricsSubmission metrics;
-            for (const auto& cache_provider : state->binary_cache_providers)
+            for (const auto& cache_provider : s.binary_cache_providers)
             {
                 auto it = metric_names.find(cache_provider);
                 if (it != metric_names.end())
                 {
                     metrics.track_define(it->second);
                 }
             }
-
-            get_global_metrics_collector().track_submission(std::move(metrics));
+
+            get_global_metrics_collector().track_submission(std::move(metrics));
+
+            s.nuget_prefix = args.nuget_id_prefix.value_or("");
+            if (!s.nuget_prefix.empty()) s.nuget_prefix.push_back('_');
+            s.use_nuget_cache = args.use_nuget_cache.value_or(false);
+            s.nuget_repo_info = get_nuget_repo_info_from_env(args);
+
+            auto& fs = paths.get_filesystem();
+            auto& tools = paths.get_tool_cache();
+            const auto& buildtrees = paths.buildtrees();
+
+            ret.nuget_prefix = s.nuget_prefix;
+
+            Optional<ZipTool> lazy_zip_tool;
+            auto zip_tool = [&]() -> const ZipTool& {
+                if (!lazy_zip_tool) lazy_zip_tool.emplace(fs, tools, stdout_sink);
+                return *lazy_zip_tool.get();
+            };
+
+            std::shared_ptr<const GcsStorageTool> lazy_gcs_tool;
+            auto gcs_tool = [&]() -> const std::shared_ptr<const GcsStorageTool>& {
+                if (!lazy_gcs_tool) lazy_gcs_tool = std::make_shared<GcsStorageTool>(tools, stdout_sink);
+                return lazy_gcs_tool;
+            };
+
+            std::shared_ptr<const AwsStorageTool> lazy_aws_tool;
+            auto aws_tool = [&]() -> const std::shared_ptr<const AwsStorageTool>& {
+                if (!lazy_aws_tool)
+                    lazy_aws_tool = std::make_shared<AwsStorageTool>(tools, stdout_sink, s.aws_no_sign_request);
+                return lazy_aws_tool;
+            };
+
+            std::shared_ptr<const CosStorageTool> lazy_cos_tool;
+            auto cos_tool = [&]() -> const std::shared_ptr<const CosStorageTool>& {
+                if (!lazy_cos_tool) lazy_cos_tool = std::make_shared<CosStorageTool>(tools, stdout_sink);
+                return lazy_cos_tool;
+            };
+            Optional<NugetBaseBinaryProvider> lazy_nuget_base;
+            auto nuget_base = [&]() -> const NugetBaseBinaryProvider& {
+                if (!lazy_nuget_base)
+                    lazy_nuget_base.emplace(
+                        fs, NuGetTool(tools, stdout_sink, s), paths.packages(), buildtrees, s.nuget_prefix);
+                return *lazy_nuget_base.get();
+            };
+
+            for (auto&& dir : s.archives_to_read)
```

Comment:
I believe this is addressed.

Reply:
Sorry, I know you like avoiding the {}s.

Maybe we could argue that they aren't necessary because we force clang-format? To be clear, I used to be anti {}s until https://www.synopsys.com/blogs/software-security/understanding-apple-goto-fail-vulnerability-2/